File: Algorithms/78. Subsets/golang/solution.go
func subsets(nums []int) [][]int {
	output := [][]int{}
	var backtrack func(int, []int)

	backtrack = func(first int, curr []int) {
		output = append(output, append([]int{}, curr...))
		for i := first; i < len(nums); i++ {
			backtrack(i+1, append(curr, nums[i]))
		}
	}

	backtrack(0, []int{})
	return output
}

File: Algorithms/78. Subsets/javascript/solution.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function (nums) {
    const backtrack = (first = 0, curr = []) => {
        output.push(curr);
        for (let i = first; i < nums.length; i++) {
            backtrack(i + 1, [...curr, nums[i]]);
        }
    };

    const output = [];
    backtrack();
    return output;
};
